src/plugins/lua/known_senders.lua

```diff
@@ -197,6 +212,55 @@ local function known_senders_callback(task)
   end
 end
 
+local function check_known_incoming_mail_callback(task)
+  local sender = task:get_reply_sender()
+  if not sender then
+    lua_util.debugm(N, task, 'Could not get sender')
+    return nil
+  end
+
+  -- making sender key
+  lua_util.debugm(N, task, 'Sender: %s', sender)
+
+  local sender_string = lua_util.maybe_obfuscate_string(tostring(sender), settings, settings.sender_prefix)
+  local sender_key = make_key_replies(sender_string:lower(), 8)
+
+  lua_util.debugm(N, task, 'Sender key: %s', sender_key)
+
+
+  local function redis_zrange_cb(err, data)
+    if err ~= nil then
+      rspamd_logger.errx(task, 'Couldn\'t get data from global replies set. Ended with error: %s', err)
+      return
+    end
+
+    rspamd_logger.infox(task, 'Successfully got list: %s of verified senders', data)
+
+    -- searching for sender in global replies set
+    if data then
+      for _, sndr in ipairs(data) do
+        if sndr == sender then
+          task:insert_result(settings.symbol_check_mail, 1.0, sndr)
+        end
+      end
+    end
+  end
+
+  -- key for global replies set
+  local global_key = make_key_replies(settings.sender_key_global, settings.sender_key_size, settings.sender_prefix)
+
+  lua_util.debugm(N, task, 'Making redis request to global replies set')
+  lua_redis.redis_make_request(task,
+          redis_params, -- connect params
+          sender_key, -- hash key
+          false, -- is write
+          redis_zrange_cb, --callback
+          'ZRANGE', -- command
+          { global_key, '0', '-1' } -- arguments
+  )
```

Comment:
Why do we need this all logic in the first place? Why not just check `global_key` for `sender`? Getting all values does not make a lot of sense to me.

Reply:
Also, I don't see any logic to verify **local** senders set. We need to check both, and the symbols might be different for each of the keys - local and global.

Reply:
I don't quite understand what you mean by checking `global_key` and verifying the local set. What should be verified in the local set and why do I need to check the `global_key` for the `sender`? This logic was created in order to get the highest score in the `global set` and sum it with the scores of incoming recipients to be sure that their scores are higher than those of new ones. It does not check anything. So, could you please explain what you mean?

Reply:
https://redis.io/docs/latest/commands/zscore/ ?